plugins/src/android/PushReceiver.java

```diff
@@ -0,0 +1,128 @@
+package com.optimove.android.cordova;
+
+import android.app.Activity;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import androidx.core.app.TaskStackBuilder;
+
+import com.optimove.android.Optimove;
+import com.optimove.android.optimobile.PushActionHandlerInterface;
+import com.optimove.android.optimobile.PushBroadcastReceiver;
+import com.optimove.android.optimobile.PushMessage;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class PushReceiver extends PushBroadcastReceiver {
+    @Override
+    protected void onPushReceived(Context context, PushMessage pushMessage) {
+        super.onPushReceived(context, pushMessage);
+        OptimoveSDKPlugin.sendMessageToJs("pushReceived", pushMessageToJsonObject(pushMessage, null));
+    }
+
+    static JSONObject pushMessageToJsonObject(PushMessage pushMessage, String actionId) {
+        JSONObject message = new JSONObject();
+
+        try {
+            message.put("id", pushMessage.getId());
+            message.put("title", pushMessage.getTitle());
+            message.put("message", pushMessage.getMessage());
+
+            if (null != pushMessage.getUrl()) {
+                message.put("url", pushMessage.getUrl().toString());
+            }
+
+            if (actionId != null) {
+                message.put("actionId", actionId);
+            }
+
+            message.put("data", pushMessage.getData());
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        return message;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void onPushOpened(Context context, PushMessage pushMessage) {
+        try {
+            Optimove.getInstance().pushTrackOpen(pushMessage.getId());
+        } catch (Exception e) {
```

Comment:
this is different from base android / flutter. There we catch `Optimobile.UninitializedException` . Better to use more specific type?

Reply:
done